### PR TITLE
Add support for Access Control by User Groups

### PR DESCRIPTION
**Pull request by memije**
### Changes:
- Added [RBAC](https://en.wikipedia.org/wiki/Role-based_access_control) middleware to support Access Control by user groups.
- Added [Casbin RBAC](https://casbin.org/docs/en/rbac) implementation.

Output of `go test ./...`:

```
❯ go test ./... --count=1
ok      github.com/protsack-stephan/gin-toolkit/httperr 0.012s
ok      github.com/protsack-stephan/gin-toolkit/httphandler     1.010s
ok      github.com/protsack-stephan/gin-toolkit/httpmod 0.009s
ok      github.com/protsack-stephan/gin-toolkit/httpmw  1.534s
```

:warning: Depends on #11

### CODE REVIEW

File: httpmw/rbac.go [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+func RBAC(fn RBACAuthorizeFunc) gin.HandlerFunc {
+	return func(c *gin.Context) {
+		if authorized, err := fn(c); err != nil || !authorized {
+			if err != nil {
+				log.Println(err)
+			}
+			httperr.Unauthorized(c)
+			c.Abort()
+			return
+		}
```
Comment (protsack-stephan) [inline httpmw/rbac.go:26]: I suggest changing the logic a little bit, let's not print the error but throw internal server error instead, something like:
```go
ok, err := fn(c)

if err != nil {
	httperr.InternalServerError(c, err.Error())
	c.Abort()
	return
}

if !ok {
	httperr.Unauthorized(c)
	c.Abort()
	return
}

c.Next()
```

File: httpmw/rbac_casbin.go [hunk 2]
```diff
@@ -0,0 +1,40 @@
+package httpmw
+
+import (
+	"errors"
+
+	"github.com/casbin/casbin/v2"
+	"github.com/gin-gonic/gin"
+)
+
+var ErrNoUser = errors.New("missing user in request context")
```
Comment (protsack-stephan) [inline httpmw/rbac_casbin.go:11]: Please add comment for this error.

File: httpmw/rbac_casbin.go [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+
+// CasbinRBACAuthorizeFunc uses a provided Casbin enforcer to implement RBAC middleware.
+// This function will look for roles stored in the gin.Context using the `groups` key
+// and will attempt to authorize the request using each found role.
+// If no match is made, the request will be rejected.
+func CasbinRBACAuthorizeFunc(e *casbin.Enforcer) RBACAuthorizeFunc {
```
Comment (protsack-stephan) [inline httpmw/rbac_casbin.go]: I would suggest changing the name a little bit, maybe `CasbinRBACAuthorizer` ? Although it makes sense to include `Func` suffix in the function type, I don't think there's much benefit to do so in the implementation.

**Review by protsack-stephan — COMMENTED** [review]
Good work! Couple small comments.

File: httpmw/rbac.go (hunk 1 above)
Comment (memije) [reply httpmw/rbac.go:26]: Makes sense. Done

**Comment by prabhat393** [comment]
Looks good.

**Pull request merged**